frame/executive/src/lib.rs

```diff
@@ -349,11 +357,27 @@ where
 	/// Execute given extrinsics and take care of post-extrinsics book-keeping.
 	fn execute_extrinsics_with_book_keeping(
 		extrinsics: Vec<Block::Extrinsic>,
+		first_signed_index: u32,
 		block_number: NumberFor<Block>,
 	) {
-		extrinsics.into_iter().for_each(|e| if let Err(e) = Self::apply_extrinsic(e) {
-			let err: &'static str = e.into();
-			panic!("{}", err)
+		extrinsics.into_iter()
+			.enumerate()
+			.for_each(|(i, ext)| {
+				// When this is true, we finished processing inherents.
+				if i as u32 == first_signed_index {
+					let mut weight = 0;
+					weight = weight.saturating_add(
+						<frame_system::Pallet<System> as OnPostInherent<System::BlockNumber>>::on_post_inherent(block_number)
+					);
+					weight = weight.saturating_add(
+						<AllPallets as OnPostInherent<System::BlockNumber>>::on_post_inherent(block_number)
+					);
+					<frame_system::Pallet::<System>>::register_extra_weight_unchecked(weight, DispatchClass::Mandatory);
```

Comment:
I think we need to have another logic branch that triggers the on_post_inherent calls in the case of no signed xts.
connects with https://github.com/paritytech/substrate/pull/9324/files#r667411177